torch/csrc/utils/tensor_new.cpp

```diff
@@ -207,6 +207,10 @@ Tensor internal_new_from_data(const Type & type, at::optional<Device> device_opt
   }
 
   if (THPVariable_Check(data)) {
+      PyErr_WarnEx(PyExc_UserWarning,
+        "To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() "
```

Comment:
you should mention `torch.as_tensor` here for the use case when you just want to get a tensor out (for data that may be mixed between e.g. tensors, numpy arrays, lists), and don't want to copy the data.

Reply:
actually I might just put warnings at `tensor_ctor` and `new_tensor` instead.